x-pack/packages/security-solution/distribution_bar/src/distribution_bar.tsx

```diff
@@ -20,56 +21,163 @@ export interface EmptyBarProps {
   ['data-test-subj']?: string;
 }
 
-const styles = {
-  base: css`
-    border-radius: 2px;
-    height: 5px;
-  `,
+const useStyles = () => {
+  const { euiTheme } = useEuiTheme();
+
+  return {
+    bar: css`
+      gap: ${euiTheme.size.xxs};
+      min-height: 7px; // for hovered bar to have enough space to grow
+    `,
+    part: {
+      base: css`
+        position: relative;
+        border-radius: 2px;
+        height: 5px;
+      `,
+      empty: css`
+        background-color: ${euiTheme.colors.lightShade};
+        flex: 1;
+      `,
+      tick: css`
+        &::after {
+          content: '';
+          opacity: 0;
+          position: absolute;
+          top: -10px;
+          right: 0;
+          width: 1px;
+          height: 6px;
+          background-color: ${euiTheme.colors.lightShade};
+        }
+      `,
+      hover: css`
+        &:hover {
+          height: 7px;
+          border-radius: 3px;
+          cursor: pointer;
+
+          .euiBadge {
+            cursor: unset;
+          }
+
+          &::after {
+            opacity: 1;
+            transition: all 0.3s ease;
+            top: -9px; // 10px - 1px to accommodate for height of hovered bar
+          }
+
+          transition: all 0.3s ease;
+        }
+      `,
+      // last tooltip should be displayed initially even without hover
+      lastTooltip: css`
+        &:last-child > div {
+          opacity: 1;
+        }
+        &:last-child::after {
+          opacity: 1;
+        }
+        &:hover ~ div:last-child > div {
+          opacity: 0;
+        }
+        &:hover ~ div:last-child::after {
+          opacity: 0;
+        }
+      `,
+    },
+    tooltip: css`
+      opacity: 0;
+      position: absolute;
+      width: 100%;
+      height: calc(
+        ${euiTheme.base + 2}px + 14px + 7px
+      ); // 2px border of the badge + 14px height of the tick + 7px height of the bar
+      text-align: right;
+      top: calc(
+        -${euiTheme.base + 2}px - 14px
+      ); // 2px border of the badge + 14px height of the tick
+      right: 0;
+
+      &:hover {
+        opacity: 1;
+        top: calc(
+          -${euiTheme.base + 2}px - 13px
+        ); // 2px border of the badge + 14px height of the tick - 1px to accomodate for height of hovered bar
+        transition: all 0.3s ease;
+      }
```

Comment:
tooltips hacks 😎

Reply:
yeah, I'm still unsure about relying on CSS quite a lot here. Maybe it would've been better to do more on the JS side. Anyway, so far it worked quite nicely, except the responsiveness problem mentioned, which I will check out and try to solve